Fix auto-creation of CALLBACK_URL for AzureAD

AparnaKarve opened the pull request:
Fix auto-creation of CALLBACK_URL for AzureAD authentication by subclassing `SocialAuthValidateCallbackMixin` in the AzureAD AuthenticatorPlugin class

Fixes https://issues.redhat.com/browse/AAP-27572

AparnaKarve committed:
Fix auto-creation of CALLBACK_URL for AzureAD

john-westcott-iv commented on test_app/tests/authentication/authenticator_plugins/test_azuread.py:
```diff
@@ -23,6 +25,45 @@ def test_azuread_auth_successful(authenticate, unauthenticated_api_client, azure
+    expected_status_code,
+    expected_error,
+):
+    callback_url = "http://testserver/api/social/complete/ansible_base-authentication-authenticator_plugins-azuread__azuread-test/"  # noqa
```
Instead of constructing this URL can we call `get_fully_qualifgied_url` like the validate method of SocialAuthValidateCallbackMixin would (or even just call validate itself)? That way if the function ever changes this test won't start failing. Because we are really testing that the URL got auto-generated and not what the content of the generated URL should be.

john-westcott-iv commented on test_app/tests/conftest.py:
```diff
@@ -80,7 +80,7 @@ def clear_content_type_cache():
 def azuread_configuration():
     return {
-        "CALLBACK_URL": "https://localhost/api/gateway/callback/github_test/",
+        "CALLBACK_URL": "https://localhost/api/gateway/callback/azuread_test/",
```
Nice find!

john-westcott-iv reviewed (approved):
Approving with minor comment.

AparnaKarve committed:
Address feedback on test

AparnaKarve commented:
@john-westcott-iv Can you please review the last commit where I have addressed your feedback?
If that looks good, I will make a similar change to the other google oauth2 PR.

Thanks.

john-westcott-iv commented on test_app/tests/authentication/authenticator_plugins/test_azuread.py:
```diff
@@ -23,6 +25,46 @@ def test_azuread_auth_successful(authenticate, unauthenticated_api_client, azure
+    if expected_error:
+        assert response.json() == expected_error
+    else:
+        slug = response.data["slug"]
+        expected_path = get_fully_qualified_url('social:complete', kwargs={'backend': slug})
+        assert response.json()['configuration']['CALLBACK_URL'] == f"http://testserver{expected_path}"
```
You can make get_fully_qualified_url auto-prepend a server by setting `FRONT_END_URL`. Something like this should work:

```suggestion
        slug = response.data["slug"]
        with override_settings(FRONT_END_URL='http://testserver/'):
            expected_path = get_fully_qualified_url('social:complete', kwargs={'backend': slug})
            assert response.json()['configuration']['CALLBACK_URL'] == expected_path
```

You would also need this import:
```
from django.test import override_settings
```

It would be bad to wrap your entire test in that as well because that should also effect the creation of the CALLBACK_URL so if the mock server changes from `http://testserver` to something else your function wouldn't care because of your override setting.

john-westcott-iv reviewed (approved):
Approving assuming tests work correctly.
